packages/superset-ui-chart/src/query/buildQueryObject.ts

```diff
@@ -12,8 +14,39 @@ function getGranularity(formData: ChartFormData): string {
 // Note the type of the formData argument passed in here is the type of the formData for a
 // specific viz, which is a subtype of the generic formData shared among all viz types.
 export default function buildQueryObject<T extends ChartFormData>(formData: T): QueryObject {
+  const extras = {
+    druid_time_origin: (formData as DruidFormData).druid_time_origin || '',
+    having: (formData as SqlaFormData).having || '',
+    having_druid: (formData as DruidFormData).having_druid || '',
+    time_grain_sqla: (formData as SqlaFormData).time_grain_sqla || '',
+    where: formData.where || '',
+  };
+
+  const orgGroupby = formData.groupby || [];
+  const orgColumns = formData.columns || [];
+  const groupbySet = new Set(orgGroupby.concat(orgColumns));
+  const limit = formData.limit ? Number(formData.limit) : 0;
```

Comment:
does `limit=0` have any weird query side effects?

Reply:
This is the original logic from `viz.py`:

`limit = int(form_data.get('limit') or 0)`